tools/rpkg/src/register.cpp

```diff
@@ -10,50 +10,31 @@
 
 using namespace duckdb;
 
-void RApi::RegisterDataFrame(SEXP connsexp, SEXP namesexp, SEXP valuesexp) {
-	if (TYPEOF(connsexp) != EXTPTRSXP) {
-		cpp11::stop("duckdb_register_R: Need external pointer parameter for connection");
-	}
-	auto conn_wrapper = (ConnWrapper *)R_ExternalPtrAddr(connsexp);
-	if (!conn_wrapper || !conn_wrapper->conn) {
+[[cpp11::register]] void rapi_register_df(duckdb::conn_eptr_t conn, std::string name, cpp11::data_frame value) {
+	if (!conn || !conn->conn) {
 		cpp11::stop("duckdb_register_R: Invalid connection");
 	}
-	if (TYPEOF(namesexp) != STRSXP || Rf_length(namesexp) != 1) {
-		cpp11::stop("duckdb_register_R: Need single string parameter for name");
-	}
-	auto name = string(CHAR(STRING_ELT(namesexp, 0)));
 	if (name.empty()) {
-		cpp11::stop("duckdb_register_R: name parameter cannot be empty");
+		cpp11::stop("duckdb_register_R: Name cannot be empty");
```

Comment:
Here and everywhere:

```suggestion
		cpp11::stop("rapi_register_df: Name cannot be empty");
```

Reply:
@nbenn can you address @krlmlr's suggestion please? Otherwise looks good to me, too.